README.md

```diff
@@ -189,168 +194,320 @@ Let's break down the example above:
+///   - environment: The environment in which to run the executable.
+///   - workingDirectory: The working directory in which to run the executable.
+///   - platformOptions: The platform specific options to use
+///     when running the executable.
+///   - output:How to handle executable's standard output
+///   - body: The custom execution body to manually control the running process
+/// - Returns a ExecutableResult type containing the return value
+///     of the closure.
+public func run<Result, Output: OutputProtocol>(
+    _ executable: Executable,
+    arguments: Arguments = [],
+    environment: Environment = .inherit,
+    workingDirectory: FilePath? = nil,
+    platformOptions: PlatformOptions = PlatformOptions(),
+    output: Output,
+    isolation: isolated (any Actor)? = #isolation,
+    body: (@escaping (Execution<Output, DiscardedOutput>, StandardInputWriter) async throws -> Result)
+) async throws -> ExecutionResult<Result> where Output.OutputType == Void
+
+/// Run a executable with given parameters and a custom closure
+/// to manage the running subprocess' lifetime, discard its
+/// standard error and write to its standard input via `StandardInputWriter`
+/// - Parameters:
+///   - executable: The executable to run.
+///   - arguments: The arguments to pass to the executable.
+///   - environment: The environment in which to run the executable.
+///   - workingDirectory: The working directory in which to run the executable.
+///   - platformOptions: The platform specific options to use
+///     when running the executable.
+///   - output:How to handle executable's standard output
+///   - body: The custom execution body to manually control the running process
+/// - Returns a ExecutableResult type containing the return value
+///     of the closure.
+public func run<Result>(
+    _ executable: Executable,
+    arguments: Arguments = [],
+    environment: Environment = .inherit,
+    workingDirectory: FilePath? = nil,
+    platformOptions: PlatformOptions = PlatformOptions(),
+    isolation: isolated (any Actor)? = #isolation,
+    body: (@escaping (Execution<SequenceOutput, DiscardedOutput>, StandardInputWriter) async throws -> Result)
+) async throws -> ExecutionResult<Result>
+
+// MARK: - Configuration Based
+/// Run a executable with given parameters specified by a `Configuration`
+/// - Parameters:
+///   - configuration: The `Subprocess` configuration to run.
+///   - output: The method to use for redirecting the standard output.
+///   - error: The method to use for redirecting the standard error.
+///   - body: The custom configuration body to manually control
+///       the running process and write to its standard input.
+/// - Returns a ExecutableResult type containing the return value
+///     of the closure.
+public func run<Result, Output: OutputProtocol, Error: OutputProtocol>(
+    _ configuration: Configuration,
+    isolation: isolated (any Actor)? = #isolation,
```

Comment:
CAn we make sure the order of the arguments stay consistent? Previously the arguments are in this order: 
- input
- output
- error
- isolation
- body

But here isolation is moved up to before output and error. Same with the one below

Reply:
Ahh good catch! Thanks!